init.sh

```diff
@@ -1,14 +1,123 @@
-# `docker ps` - copy container ID and replace [container ID] in `docker stop [411b2c4fd3e9]` then `docker rm [411b2c4fd3e9]`
-docker run --name honeydo-db -e POSTGRES_DB=honeydo -e POSTGRES_USER=honeydo-user -e POSTGRES_PASSWORD=mysecretpassword -p 5432:5432 -d postgres
-dotnet tool install --global ef #will warn if alaready installed
+## Setup a postgres docker and migrate the tables to it - if this fails, run the following commands to remove the container and re-create it
+## troubleshooting: `docker ps` - copy container ID and replace [container ID] in `docker stop [411b2c4fd3e9]` then `docker rm [411b2c4fd3e9]`
+echo "Checking for docker container."
+if docker container ls -a | grep 'honeydo-db'; then
+    echo "Docker has a container ready. Checking if container is running."
+    if docker ps | grep 'honeydo-db'; then
+        echo "Docker container is already running. Skipping..."
+    else
+        echo "Starting container..."
+        docker start honeydo-db
+    fi
+else 
+    echo "Creating docker postgres image and migrating the database."
+    docker run --name honeydo-db -e POSTGRES_DB=honeydo -e POSTGRES_USER=honeydo-user -e POSTGRES_PASSWORD=mysecretpassword -p 5432:5432 -d postgres
+fi
+
+## Install dotnet ef for db migration
+echo "Checking for dotnet tool 'ef'."
+if dotnet ef --version; then
+    echo "'ef' is installed."
+else
+    echo "Installing dotnet tool 'dotnet-ef'."
+    dotnet tool install --global dotnet-ef #will warn if alaready installed
+fi
+
+## Database migration TODO: Set flag to only do database migration
+echo "Migrating the database on docker"
 dotnet ef database update -p ./HoneyDo.Infrastructure/ -s ./HoneyDo.Web/ -c HoneyDoContext
 
-# Install dotnet dev-certs for https and then trust them
-dotnet tool install --global dotnet-dev-certs #will warn if already installed
-dotnet dev-certs https # linux
-dotnet dev-certs https --trust # mac/windows
+## Install dotnet dev-certs for https
+echo "Checking for dotnet tool 'dev-certs'."
+if dotnet dev-certs --version; then
+    echo "'dev-certs' is installed."
+else
+    echo "Installing dotnet tool 'dotnet-dev-certs'."
+    dotnet tool install --global dotnet-dev-certs #will warn if already installed
+fi
+
+## Setting dotnet to trust ssl certificates for development
+echo "Trusting development ssl certificates..."
+if [ "$(expr substr $(uname -s) 1 5)" == "Linux" ]; then
+    dotnet dev-certs https # linux
+else
+    dotnet dev-certs https --trust # mac/windows
+fi
+
+## Set up JwtKey with input or random key with dotnet-user-serets
+echo "Checking for dotnet tool 'user-secrets'."
+if dotnet user-secrets --version; then
+    echo "'user-secrets' is installed."
+else
+    echo "Installing dotnet tool 'dotnet-user-secrets'..."
+    dotnet tool install --global dotnet-user-secrets
+fi
+
+## Set up JwtKey with defined key or random key TODO: Ignore it
+read -p "Input JwtKey or press (enter) for a random key:" key
```

Comment:
The key has to be longer than 16 characters, no need to check if it isn't but we should echo that to the user.

Reply:
I have to leave a comment to mark for `Requested Changes`

Reply:
I apparently didn't notified you had reviewed the PR so I'll go through that.